src/facade/redis_parser.cc

```diff
@@ -59,17 +57,18 @@ auto RedisParser::Parse(Buffer str, uint32_t* consumed, RespExpr::Vec* res) -> R
         LOG(FATAL) << "Unexpected state " << int(state_);
     }
 
-    *consumed += last_consumed_;
+    *consumed += resultc.second;
 
-    if (last_result_ != OK) {
+    if (resultc.first != OK) {
       break;
     }
-    str.remove_prefix(last_consumed_);
-  }
+    str.remove_prefix(resultc.second);
+    resultc.second = 0;
```

Comment:
`str.remove_prefix(std::exchange(resultc.second, 0));`

Reply:
It's a preference -- I don't care if you apply this or not just commenting out that there `exchange` exists :nerd_face: